### PR TITLE
cloud test script selection

### PR DESCRIPTION
**Pull request by Llandy3d**
This adds the possibility to select a script from a list of cloud tests on your account for the current project id.

Presentation needs to drastically be improved but the logic is there! 😄

### CODE REVIEW

File: src-tauri/src/cloud.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+fn get_request(url: &str, cloud_token: &str) -> reqwest::RequestBuilder {
+
+    // NOTE: messy but will do
+    let client = reqwest::Client::new();
+    let url = format!("https://api.k6.io{}", url);
+    client.get(url)
+        .header(reqwest::header::AUTHORIZATION, format!("Token {}", cloud_token))
+        .header(reqwest::header::USER_AGENT, "kroco6")
+}
```
Comment (Llandy3d) [inline src-tauri/src/cloud.rs:11]: I was planning to use the same client/headers setup for multiple cloud requests in the future, I went as far as letting it understand the type you wanted as to handle the complete request inside of it but I encountered issues not worth pursuing at this stage... so it just creates the best request & adds headers for now.

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -56,14 +60,33 @@ fn main() {
...
+    let project_config = state
+        .project_manager
+        .load_project_config(project)
+        .map_err(|e| e.to_string())?;
+
+    if project_config.cloud_token.is_none() || project_config.cloud_project_id.is_none() {
+        return Err("missing cloud_token/cloud_project_id config".to_string());
+    }
+    let cloud_token = project_config.cloud_token.unwrap();
+    let cloud_project_id = project_config.cloud_project_id.unwrap();
```
Comment (Llandy3d) [inline src-tauri/src/main.rs:74]: I'm sure there is a better to describe this, but i wasn't able to remember 🤷‍♂️

File: src-tauri/src/main.rs [hunk 3]
```diff
@@ -56,14 +60,33 @@ fn main() {
...
 async fn close_splashscreen(window: Window) {
     // Close splashscreen
-    window
-        .get_window("splashscreen")
-        .expect("no window labeled 'splashscreen' found")
-        .close()
-        .unwrap();
+    if let Some(window) = window.get_window("splashscreen") {
+        window.close().unwrap();
+    }
```
Comment (Llandy3d) [inline src-tauri/src/main.rs:89]: changed to check to see if the splashscreen is there before closing, encountered some race conditions or script reloads 🤷‍♂️

File: src/routes/test/edit/ScriptEditor.svelte [hunk 4]
```diff
@@ -8,15 +8,34 @@
...
   let container: HTMLDivElement;
   let editor: monaco.editor.IStandaloneCodeEditor;
 
   let script = "";
+  let cloud_tests: Array<CloudTest> = [];
+  let cloudTestDialogOpen = false;
+
+  async function loadCloudTests() {
+    // TODO: we need to know the current active project
+    const project: Project = { name: "default", test_collections: [] };
```
Comment (Llandy3d) [inline src/routes/test/edit/ScriptEditor.svelte:29]: same thing as the other PR until we have shared project state

File: src/routes/test/edit/ScriptEditor.svelte [hunk 5]
```diff
@@ -98,5 +117,32 @@
...
+            {/each}
+            </div>
+          This action cannot be undone. This will permanently delete your account
+          and remove your data from our servers.
+        </Dialog.Description>
+      </Dialog.Header>
+    </Dialog.Content>
+  </Dialog.Root>
+    <div class="full-w flex-auto" bind:this={container}></div>
+  </div>
```
Comment (Llandy3d) [inline src/routes/test/edit/ScriptEditor.svelte:148]: don't judge me, I just wanted to it to be visible 😄

**Review by oleiade — APPROVED** [review]
LGTM 🦖 🦕 🚀  (Insert jurassic park's music here)

**Pull request merged**